src/main/kotlin/no/nav/syfo/infrastructure/database/repository/ForesporselRepository.kt

```diff
@@ -0,0 +1,78 @@
+package no.nav.syfo.infrastructure.database.repository
+
+import no.nav.syfo.application.IForesporselRepository
+import no.nav.syfo.domain.Foresporsel
+import no.nav.syfo.domain.Personident
+import no.nav.syfo.domain.Veilederident
+import no.nav.syfo.domain.Virksomhetsnummer
+import no.nav.syfo.infrastructure.database.DatabaseInterface
+import no.nav.syfo.infrastructure.database.toList
+import java.sql.ResultSet
+import java.time.OffsetDateTime
+import java.util.*
+
+class ForesporselRepository(val database: DatabaseInterface) : IForesporselRepository {
+    override fun createForesporsel(foresporsel: Foresporsel): Foresporsel {
+        return database.connection.use { connection ->
+            val pForesporsel =
+                connection.prepareStatement(CREATE_FORESPORSEL).use {
+                    it.setString(1, foresporsel.uuid.toString())
+                    it.setObject(2, foresporsel.createdAt)
+                    it.setString(3, foresporsel.arbeidstakerPersonident.value)
+                    it.setString(4, foresporsel.veilederident.value)
+                    it.setString(5, foresporsel.narmestelederPersonident.value)
+                    it.setString(6, foresporsel.virksomhetsnummer.value)
+                    it.executeQuery().toList { toPForesporsel() }.single()
+                }
+            connection.commit()
+            pForesporsel.toForesporsel()
+        }
+    }
+
+    override fun getForesporsel(foresporselUuid: UUID): Foresporsel? {
```

Comment:
Lurer litt på om denne bare burde vært `getForesporsler(): List<Foresporsel>` i stedet, vet ikke om jeg kommer på spesifikke usecases likevel for å bruke denne. Dette blir vel mer hvis man allerede har en Foresporsel-uuid man ønsker å gjøre en endring på, feks behandle/lukke eller lignende 🤔

Reply:
Hmja, eller `getForesporsler(personident): List<Foresporsel>` ? Det blir vel det mest akuelle use-case vil jeg tro.

Reply:
Ja riktig 👍🏼 Tror jeg endrer til det.